lib/granite/action/preconditions.rb

```diff
@@ -105,8 +106,15 @@ def decline_with(*args, **kwargs)
 
       private
 
-      def run_preconditions!
-        _preconditions.execute! self
+      def run_preconditions!(cache_result: false)
```

Comment:
Are there any reasons to not cache results for all calls of `run_preconditions!`?

Reply:
I tried ~10 different approaches to the problem, and this is the only one that doesn't break dozens of specs on platform. 
Granite is a mess, and platform depends on the mess. I have ideas on how to untangle that for the next major release, but for now, this is a duct-tape solution. It's the only one we can get, unfortunately

Reply:
I see how it could be difficult to solve, let's try with that approach and see how it goes.

Reply:
Just for the record: I absolutely would like to have a cleaner solution in here, and will try to get us there. Definitely what I gathered from trying to solve this, is that we need to start with a cleanup and untangling the spaghetti.